### PR TITLE
fix: Remove negative margin from Button inline link variant

### PR DESCRIPTION
**Pull request by michaeldowseza**
### Description

Removed negative margin for inline link variant. Expected visual difference in Button permutations.

Before:
Screenshot 2024-04-20 at 13 14 42

After:
Screenshot 2024-04-20 at 13 14 58

Related links, issue #, if available: n/a
AWSUI-37538

### How has this been tested?

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by timogasda** [comment]
Is "Icon margin escaping Button inline link variant container" the best description for the release notes? I would be more clear with what we're fixing.

File: src/button/styles.scss [hunk 1]
```diff
@@ -147,6 +147,19 @@
     margin-inline-start: awsui.$space-xxs;
   }
 
+  &.variant-inline-link {
+    > .icon-left {
+      inset-inline-start: 0;
```
Comment (timogasda) [inline src/button/styles.scss:152]: Why is this needed?

File: src/button/styles.scss [hunk 2]
```diff
@@ -147,6 +147,19 @@
...
+      margin-inline-end: awsui.$space-xs;
+    }
+
+    > .icon-right {
+      position: relative;
```
Comment (johannes-weber) [inline src/button/styles.scss:157]: The position property is not needed - it's inherited from https://github.com/cloudscape-design/components/pull/2081/files#diff-6529fe18ff5331f59396707323f6469db1d2002399ab5e95515ab2c43164889aR144

File: src/button/styles.scss (hunk 1 above)
Comment (johannes-weber) [reply src/button/styles.scss:152]: Is there a reason whey we change it exclusively for the inline variant and not for all button variants with anicon?

Comment (michaeldowseza) [reply src/button/styles.scss:152]: This is the only variant that is actually affected by this. Normally a button has an outline around it and so the icon is shifting within that outline. If this is applied to all icons in a button then the gaps between the edge of the button and the icon and the gap between the end of the text and the opposite end of the button are not even.

**Comment by michaeldowseza** [comment]
> Is "Icon margin escaping Button inline link variant container" the best description for the release notes? I would be more clear with what we're fixing.

Any suggestions? I thought this was exactly what this was fixing?

File: src/button/styles.scss [hunk 3]
```diff
@@ -147,6 +147,19 @@
...
+      inset-inline-end: 0;
+      margin-inline-start: awsui.$space-xs;
+    }
```
Comment (johannes-weber) [inline src/button/styles.scss:160]:
```suggestion
    > .icon-right {
      inset-inline-end: 0;
      margin-inline-start: awsui.$space-xs;
    }
```

The position property is not needed - it's inherited from https://github.com/cloudscape-design/components/pull/2081/files#diff-6529fe18ff5331f59396707323f6469db1d2002399ab5e95515ab2c43164889aR144

**Comment by johannes-weber** [comment]
> > Is "Icon margin escaping Button inline link variant container" the best description for the release notes? I would be more clear with what we're fixing.
> 
> Any suggestions? I thought this was exactly what this was fixing?

What about: "Realign icon margin for Button inline link variant"?

**Comment by timogasda** [comment]
(ugh, why can't I do top-level threads in GitHub?!)

I don't have any suggestion for the PR title, because I still don't really understand what we're trying to fix 😅

File: src/button/styles.scss (hunk 3 above)
Comment (michaeldowseza) [reply src/button/styles.scss]: Thanks, removed.

**Review by johannes-weber — APPROVED** [review]
